specification/dartLangSpec.tex

```diff
@@ -12483,9 +12485,10 @@ \subsubsection{Unqualified Invocation}
 
 \commentary{%
 This occurs when the lexical lookup has determined that
-$i$ must invoke an instance member,
+$i$ must invoke an instance member of a class or an extension,
 and the location of $i$ can access \THIS{},
-and the interface of the enclosing class has a member named \id.
+and the interface of the enclosing class has a member named \id,
```

Comment:
Why do we bother to check and assert this here?  As best I can tell, section \ref{ordinaryInvocation} which we delegate to does this check anyway.  So it seems like we should just not talk about this at all.  That is, we should say "When lexical lookup fails to find a declaration, i is treated as this.i".  And then if i is not in the interface for this, then ordinaryInvocation will handle it just fine.  What am I missing?

Reply:
I'm just explaining exactly what we know to be true here. I could delete the whole commentary.

However, we can't treat `i` as `this.i` if we found a setter, but there was no getter. In that case actually did not find any declarations named `i` (OK, there could have been a getter in an enclosing scope, but if we find a setter then we insist on using the getter from the same scope), so "When lexical lookup fails to find a declaration" must be interpreted as "When lexical lookup did not encounter a compile-time error, and it did not find a declaration named `i`".

I think it's simpler to determine precisely when we need to prepend `this`, and then communicate that fact unambiguously to the 'caller'.